include/mbedtls/ssl.h

```diff
@@ -49,6 +49,15 @@
 #endif
 
 #if defined(MBEDTLS_ZLIB_SUPPORT)
+
+#if defined(MBEDTLS_DEPRECATED_WARNING)
+#warning "Record compression support via MBEDTLS_ZLIB_SUPPORT is deprecated and will be removed in the next major revision of the library"
```

Comment:
I don't like being so precise about when it will be removed. That's a commitment that we don't need to make. I'd rather stay with “will be removed in a future version of the library”.

Reply:
I'm comfortable with the wording. We can always decide to keep it in for longer if we wish.

The key thing here, is if we're going to remove it, people need certainty. It's better to surprise them by removing it later than we said, than earlier than we said. Being vague isn't very helpful either, as people won't take us seriously enough.